feat: Visual Metrics Editing

briangregoryholmes opened the pull request.

ericpgreen2 commented on web-common/src/features/visual-metrics-editing/MetricsTable.svelte:
```diff
@@ -0,0 +1,262 @@
+    const initialFormatWidth =
+      type === "measures" ? cells[4].getBoundingClientRect().width : 0;
+
+    nameWidth.set(initialNameWidth);
+    labelWidth.set(initialLabelWidth);
+    formatWidth.set(initialFormatWidth);
```
I might be missing it but I don't see these stores set again, so I'm wondering if the `MaxStore` layer is needed?

briangregoryholmes replied:
They are all set once per table, so twice for each store.

ericpgreen2 reviewed (approved):
Overall looks good! Given it's behind a feature flag, I'd like to merge and we can work on the final mile in follow-on PRs. In the final mile, the feature as a whole probably merits a dedicated e2e test.

Here are some UXQA findings:
1. I'm unable to submit my edits for any of the measure input fields, like `name`: [Jam](https://jam.dev/c/fc4c2ff3-56c6-47a6-986a-366e1ceec1d1)
2. Leftover "true" text: 
    <img width="296" alt="image" src="https://github.com/user-attachments/assets/25a33aee-afd9-4005-9181-9fae533b5dbe">
3. Given point 1, I wasn't able to fully test it, but we should make sure that `count(*)` is allowed
    <img width="316" alt="image" src="https://github.com/user-attachments/assets/aaafb179-d313-4fcf-b502-01f03957df31">
4. I think "Summable metric" should default to `true`
5. @jkhwu can we add some space between the Measures & Dimensions tables? They feel uncomfortably close.
6. I used the dropdown to switch to a new model, and I hit the following error. When using the model switcher, can we remove the `table` key (if any) from the YAML?
    <img width="1242" alt="image" src="https://github.com/user-attachments/assets/e5663ee1-9101-4033-957b-858533218c9d">
7. We should store the view state (code or viz) somewhere (local storage?) so that refreshing the code view doesn't automatically flip you to the viz view. Similarly, navigating from code->Preview->"Edit metrics" should take me back to code.

briangregoryholmes commented on web-common/src/features/canvas/ViewSelector.svelte:
```diff
@@ -11,45 +11,48 @@
 
-  export let selectedView = "split";
+  export let selectedView: "code" | "split" | "viz" = "split";
+  export let split = true;
```
Adjusted

briangregoryholmes commented:
1. You are missing Format, which is a required field. This is one of the reasons why I think inline editing is necessary for this surface (mismatch between UI and YAML requirements), but I've added a tooltip to the save changes button that lists out missing required fields. 
2. Removed
3. It works.
4. Agreed. Changed.
5. My spacing was off from Figma. Updated.
6. Fixed.
7. Working on it.

jkhwu commented:
Note to self: 
`features:
  visualEditing: true`
